### ISSUE TITLE
Gamespot image viewers 

### ISSUE THREAD

**Issue by tsunami2311** [gorhill/uBlock#2835]
So i found out so long and ublock origin is ENABLED even IF you whitelist the whole site of gamespot its images are not clickable will not open the site image viewer what so ever, only thing that restore it is out right disabling ublock origin on the site, this same issue is causing forum posting and commenting to not work either.

https://www.gamespot.com/sundered/images/
https://www.gamespot.com/articles/final-fantasy-15s-july-update-is-out-now-with-dela/1100-6452132/#comments-block-33400378

Seeing Gamespot has track record of out right  breaking there site for anyone even have a blocker enabled i think it them doing it purposely but I fiqure i post it here to see if it is them or ublock origin cause other are reporting they same issue using the same blocker

EasyPrivacy
Malvertising filter list
Malware Domain List​​​​​
​​Malware domains​
​​​​Malware domains (long-lived)​
Malware filter list by Disconnect​​​​​
Fanboy’s Social Blocking List
MVPS HOSTS​​​​​12,598
​​Peter Lowe’s Ad and tracking server list​

Like said i already disabled all these and even whitelisted the site temporally  and doing so does not restore function only hitting the power button and refreshing restores function so i dont think is due to the filters less i missing something

    ​​

**Comment by gorhill** [comment]
Unable to reproduce any issue with default filter settings/lists.

Using the logger is _mandatory_ if you want to make the case this is not a filter issue.

Also, you ignored the information asked from you in the template.

Because of this, closing and marking as _invalid_, as per [CONTRIBUTING](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md).

**Comment by okiehsch** [comment]
I can reproduce the issue, an easier way to reproduce is to go to
`https://www.gamespot.com/images/1300-3260499/`. the image will not load on my end.
````
gamespot.com#@#script:inject(abort-on-property-read.js, _bt)
@@||static.gamespot.com/js/compiled/c27ebcd.js$script,first-party
````
fixes this issue.

**Comment by gorhill** [comment]
Ok, reproduced.

Disabling uBO **does** fix the issue. However whitelisting all requests does not, which is hard to explain.

**Comment by gorhill** [comment]
@okiehsch I see, unless uBO is fully disabled, the scriplets will still be injected.

Unfortunately the scriplets are still not reported in the logger, so there was no clue as to what was happening.

Thinking about it, whitelisting `gamespot.com` should prevent the scriptlets from being injected, since they are injected as inline script tags, i.e. first-party.

**Comment by okiehsch** [comment]
> unless uBO is fully disabled, the scriplets will still be injected.

If I
1. Add theguardian.com##script:inject(abort-on-property-write.js, Math.floor) to my filters.
2. Go to `https://www.theguardian.com/international`,  some site breakage will happen ass expected,
for example clicking on search will redirect me to `https://www.google.co.uk/advanced_search?q=site:www.theguardian.com` and the console will note the `Math.floor` error.
3. If I now whitelist the `theguardian.com` and reload, search will work as expected and no Math.floor
error in the console.

So whitelisting does lead to the disabling of scriptlets, atleast in some cases.

**Comment by gorhill** [comment]
> If I now whitelist the `theguardian.com` and reload, search will work as expected and no Math.floor
error in the console.

By "whitelist" you mean setting an `allow` rule for `theguardian.com`? If so, does not work on my side without the fix above.

**Comment by gorhill** [comment]
Ok sorry, by "fully disabled" I meant disabling through the power button, not disabling the extension itself.

**Comment by okiehsch** [comment]
I meant clicking the blue power button or manually adding `theguardian.com` to the Whitelist-tab.

**Comment by tsunami2311** [comment]
gamespot.com#@#script:inject(abort-on-property-read.js, _bt)
@@||static.gamespot.com/js/compiled/c27ebcd.js$script,first-party

Sweet this fixes the image viewer and comment/forum issue, atlest till Gamespot start play we dont alike adlockers game and purpose breaks things if this problem was gamespot ?

**Comment by gorhill** [comment]
@tsunami2311 Sorry for dismissing your issue too quickly, this was an actual issue with uBO.

**Comment by tsunami2311** [comment]
all good like i been blame gamespot for there endless issue with there site  with adblockers so i fiqure i post it here just in case it was wasn't sorry for lack of info

**Comment by tsunami2311** [comment]
out of curiousty this fix your commited will fix this IF i whitelist the site? or will it work with out whitelisting the site which i dont want to do

**Comment by tsunami2311** [comment]
after more testing I find that  fixes the Picture viewer, but this fix will revert the Quality player to the shitty ad player

> @@||static.gamespot.com/js/compiled/c27ebcd.js$script,first-party

this will fix comment and forum posting, and keep  the qualitly player instead of the  shitty ad player 

> 
gamespot.com#@#script:inject(abort-on-property-read.js, _bt)

**Comment by tsunami2311** [comment]
sigh gamespot fixed the posting/commenting issue them selves but the above fixes also no long fix there image viewer, once again ublock has to disabled for image viewer to work

**Comment by gorhill** [comment]
My fix was silly, I didn't get that `/sundered/` was just for one specific game.

**Comment by tsunami2311** [comment]
sorry if i am being pain, it just i lost hope in gamespot haveing function web site long ago, is this fix build update or filter update?  and i will  test again

**Comment by gorhill** [comment]
It's a filter list update, force an update of "uBlock filters".

**Comment by tsunami2311** [comment]
https://www.gamespot.com/pyre/images/
https://www.gamespot.com/sundered/images/

 works now   one last issue if you go to any page really  with any game images in or image in general example
https://www.gamespot.com/reviews/sundered-review/1900-6416719/

IF you click on any of images on the page or any other page likes this it should also open the image viewer like the above links do now, it does not though,


Gamespot is really terrible about breaking there site with blockers used

**Comment by gorhill** [comment]
I need to investigate, not sure there is a solution. If I do not inject the scriptlet on that page, the image viewer works but the video has a pre-roll ad. I need to investigate deep in the code to find out what is exactly is preventing the viewer to work when the scriptlet is active.

**Comment by tsunami2311** [comment]
ok just post again when you think have solution, im convinced gamespot purposely does this cause i dont remember a time when gamespot site DIDNT have endless issue with blockers just remeber it not just that page is any page that images  that have this happen, if go to articles/reveiws actual image url it works fine

If  you can find way to fix this all and keep good video play great, if not i use to get the bad player with ads and no quality control.

I should of stoped using gamespot long ago

**Comment by tsunami2311** [comment]
just forced update on the ublock filters and  image viewer is working on the whole site again, commenting and forums posts are working and it still keeps the quality Video play as far as i can tell. 

So this seems all fixed again, atlest till gamespot start screw around again, out curioustiy was the issue with ublock origin or with what gamespot was doing purposely to break there site?

**Comment by gorhill** [comment]
Just needed to work on filters. If it happens again, open an issue at uAssets.

The uBO issue, not being able to prevent scriptlet from being applied has been fixed, and the fix is part of 1.13.9 beta.

**Comment by tsunami2311** [comment]
will do and thinks again you made alot people frustrated with gamespot happy for time being
